nemoguardrails/library/domain_hallucination/verification.py

```
@@ -0,0 +1,712 @@
+    else:
+        urls = [f"https://{target}/", f"http://{target}/"]
+
+    last: Dict[str, Any] = {}
+    for url in urls:
+        from urllib.parse import urlparse
+
+        parsed = urlparse(url)
+        domain = parsed.hostname or ""
+        started = time.perf_counter()
+
+        request = Request(url, method="HEAD", headers={"User-Agent": "DomainGuard/0.1"})
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)